astronomer/providers/amazon/aws/hooks/s3.py

```diff
@@ -114,15 +116,17 @@ async def list_prefixes(
         return prefixes
 
     @provide_bucket_name_async
-    async def get_file_metadata(self, client: AioBaseClient, bucket_name: str, key: str) -> List[Any]:
+    async def get_file_metadata(
+        self, client: AioBaseClient, bucket_name: str, key: str | None = None
+    ) -> list[Any]:
         """
         Gets a list of files that a key matching a wildcard expression exists in a bucket asynchronously
 
         :param client: aiobotocore client
         :param bucket_name: the name of the bucket
         :param key: the path to the key
         """
-        prefix = re.split(r"[\[\*\?]", key, 1)[0]
+        prefix = re.split(r"[\[\*\?]", key, 1)[0] if key else ""
```

Comment:
Should we set the default value of `key` to "" instead of adding an if here? Or is there anything I missed?

Reply:
yes, this make sense

Reply:
Ah yes, I misunderstood this initialization as well 🤦‍♂️ Thanks for clarification